### ISSUE TITLE
Unable to Specify Robot Name and Modify Port for Extern Controllers in Webots R2023b on Windows 10

### ISSUE THREAD

**Issue by ningtianze** [cyberbotics/webots#6418]
I am working in a world setup where I have a robot node (Abot) and a supervisor node (robot) in Webots R2023b on a Windows 10 system. According to the user guidelines, I am trying to control these two nodes on the same port (1234) under different processes, as well as under different ports. Taking the Abot node as an example, I use the following command:
```
command = [
        "D:\\Program Files\\Webots\\msys64\\mingw64\\bin\\webots-controller.exe",
        "D:\\Abot\\controllers\\robot_run\\robot_run.py",
        "--robot-name='Abot'",
        ]
```
However, no matter how I attempt, when both nodes have their controller set to extern, the result always displays:
`The started controller targets a local instance (ipc protocol) of Webots with port number 1234. Targeting the only robot waiting for an extern controller.
No robot name provided, Webots instance 1234 should have exactly one robot set with an <extern> controller.
Available robots with <extern> controllers are:
 * Abot
 * robot`

I am puzzled and tried the same command on a Windows CMD window and got the same result:+

`C:\Users\ningtianze>"D:\\Program Files\\Webots\\msys64\\mingw64\\bin\\webots-controller.exe" "D:\\Abot\\controllers\\robot_run\\robot_run.py" "--robot-name='Abot'"

The started controller targets a local instance (ipc protocol) of Webots with port number 1234. Targeting the only robot waiting for an extern controller.
No robot name provided, Webots instance 1234 should have exactly one robot set with an <extern> controller.
Available robots with <extern> controllers are:
 * Abot
 * robot
`
However, when only one extern is opened in the world, the command webots-controller.exe [options] path/to/controller/file [controller-args] can function normally, and the --robot-name can be ignored.

In summary, the [controller-args] part mentioned in the [Webots documentation](https://cyberbotics.com/doc/guide/running-extern-robot-controllers?tab-os=windows&tab-language=python) seems to have no effect.

I want to know how to effectively modify the port and control based on the robot name.

My issue is similar to #4680, which also hasn't been resolved.

Background:
I am using Windows 10, PyCharm, and Webots R2023b.

I am submitting this issue to seek help on how to specify the robot name and modify the port for extern controllers in my setup.

**Comment by zed-don** [comment]
I have the same question,have you soloved it?

**Comment by ningtianze** [comment]
yes, Verify that the environment variables on your computer are set correctly,  ---> path

**Comment by zed-don** [comment]
thanks but I havn't solve my problem,I will try more solutions.

**Comment by ningtianze** [comment]
os.environ["WEBOTS_HOME"] = r"D:\Program Files\Webots"
    os.environ["Path"] += r"D:\Program Files\Webots\lib\controller\;" + \
                          r"D:\Program Files\Webots\msys64\mingw64\bin\;" + \
                          r"D:\Program Files\Webots\msys64\mingw64\bin\cpp\;"
    webots_cmd_line = ["webotsw.exe", filepath]
    controller_file_cmdline = ["webots-controller.exe",
                               f"--robot-name={name}",
                               filepath]
    maybe you can try this operation，
